hldb/Lib/DirectConnector.swift

```diff
@@ -9,24 +9,460 @@
+        
+        for colIdx in 0..<columnCount {
+          let columnName: String = String.fromCString(sqlite3_column_name(statement, colIdx))!
+          print(columnName)
+          let objectValue = self.objectForColumnIndex(Int(colIdx));
+          dict.setValue(objectValue, forKeyPath: columnName)
+        }
+        
+        return dict;
+      } else {
+        NSLog("Warning: There seem to be no columns in this set.");
+        return nil;
+      }
+      
+    } else {
+      return nil
+    }
+  }
+  
+  func objectForColumnIndex(colIndex: Int) -> AnyObject? {
+    let colIdx = Int32(colIndex)
+    let columnType = Int32(sqlite3_column_type(statement, colIdx));
+    
+    let returnValue: AnyObject;
     
+    if (columnType == SQLITE_INTEGER) {
+      returnValue = NSNumber(longLong: sqlite3_column_int64(statement, colIdx))
+    } else if (columnType == SQLITE_FLOAT) {
+      returnValue = NSNumber(double: sqlite3_column_double(statement, colIdx))
+    } else if (columnType == SQLITE_BLOB) {
+      
+      let bytes = sqlite3_column_blob(statement, colIdx)
+      let length = Int(sqlite3_column_bytes(statement, colIdx))
+      returnValue = Blob(bytes: bytes, length: length)
+    } else {
+      //default to a string for everything else
+      returnValue = String.fromCString(UnsafePointer<CChar>(sqlite3_column_text(statement, colIdx)))!
+    }
+    
+    return returnValue;
   }
   
-  func openDatabase() -> COpaquePointer? {
+}
+
+
+
+public class DirectDB: AbstractDB {
+  
+  public typealias Cursor = DirectCursor
+  
+  //private let dbPath = "DirectDB"
+  
+  internal let dbPointer: COpaquePointer?
+  private var inTransaction: Bool = false
+  private var isExecutingStatement = false
+  private let loggingErrors = true
+  private var errorMessage: String = ""
+  
+  let SQLITE_TRANSIENT = unsafeBitCast(-1, sqlite3_destructor_type.self)
+  
+  public init(dbPath: String) {
     var db: COpaquePointer = nil
-    let path = "foo"
-    if sqlite3_open(path, &db) == SQLITE_OK {
-      print("Successfully opened connection to database at \(path)")
-      return db
+    if sqlite3_open(dbPath, &db) == SQLITE_OK {
+      self.dbPointer = db
+      print("Database opened!!")
     } else {
       print("Unable to open database. Verify that you created the directory described " +
         "in the Getting Started section.")
-//      XCPlaygroundPage.currentPage.finishExecution()
+      self.dbPointer = nil
     }
-    return nil
   }
   
+  public func executeUpdate(sql:String) -> Bool {
+    return self.executeUpdate(sql, withArgumentsInArray: [])
+  }
+  
+  public func executeUpdate(sql:String, withArgumentsInArray arrayArgs:[AnyObject]) -> Bool {
+    
+    print(sql)
+    print(arrayArgs)
+    
+    guard let _ = self.dbPointer else { return false }
+    
+    if (self.isExecutingStatement) {
+      NSLog("WARNING: DirectDB already executing statement")
+      return false;
+    }
+    
+    self.isExecutingStatement = true
```

Comment:
I think this should be okay assuming it's on already on a serial queue, by which I mean you can remove this